examples/timeseries/event_classification_for_payment_card_fraud_detection.py

```diff
@@ -0,0 +1,430 @@
+
+# Load the dataset as a Pandas dataframe.
+cache_path = "fraud_detection_cache.csv"
+if not os.path.exists(cache_path):
+    print("Download dataset")
+    dataframes = []
+    num_files = (end_date - start_date).days
+    counter = 0
+    while start_date <= end_date:
+        if counter % (num_files // 10) == 0:
+            print(f"[{100 * (counter+1) // num_files}%]", end="", flush=True)
+        print(".", end="", flush=True)
+        url = f"https://github.com/Fraud-Detection-Handbook/simulated-data-raw/raw/main/data/{start_date}.pkl"
+        dataframes.append(pd.read_pickle(url))
+        start_date += datetime.timedelta(days=1)
+        counter += 1
+    print("done", flush=True)
+    transactions_dataframe = pd.concat(dataframes)
+    transactions_dataframe.to_csv(cache_path, index=False)
+else:
+    print("Load dataset from cache")
+    transactions_dataframe = pd.read_csv(
+        cache_path, dtype={"CUSTOMER_ID": bytes, "TERMINAL_ID": bytes}
+    )
+
+print(f"Found {len(transactions_dataframe)} transactions")
+
+"""
+Each transaction is represented by a single row, with the following columns of interest:
+
+- **TX_DATETIME**: The date and time of the transaction.
+- **CUSTOMER_ID**: The unique identifier of the customer.
+- **TERMINAL_ID**: The identifier of the terminal where the transaction was made.
+- **TX_AMOUNT**: The amount of the transaction.
+- **TX_FRAUD**: Whether the transaction is fraudulent (1) or not (0).
+"""
+
+transactions_dataframe = transactions_dataframe[
+    ["TX_DATETIME", "CUSTOMER_ID", "TERMINAL_ID", "TX_AMOUNT", "TX_FRAUD"]
+]
+
+transactions_dataframe.head(4)
+
+"""
+The dataset is highly imbalanced, with the majority of transactions being legitimate.
+"""
+
+fraudulent_rate = transactions_dataframe["TX_FRAUD"].mean()
+print("Rate of fraudulent transactions:", fraudulent_rate)
+
+"""
+The [pandas dataframe](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.html) is converted into a [Temporian event-set](https://temporian.readthedocs.io/en/latest/reference/temporian/EventSet/), which is better suited for the data exploration and feature preprocessing of the next steps.
+"""
+
+transactions_eventset = tp.from_pandas(
+    transactions_dataframe, timestamps="TX_DATETIME"
+)
+
+transactions_eventset
+
+"""
+It is possible to plot the entire dataset, but the resulting plot will be difficult to read. Instead, we can group the transactions per client.
+"""
+
+transactions_eventset.add_index("CUSTOMER_ID").plot(indexes="3774")
+
+"""
+Note the few fraudulent transactions for this client.
+
+## Preparing the training data
+
+Fraudulent transactions in isslation cannot be detected. Instead, we need to connect related transactions. For each transaction, we compute the sum and count of transactions for the same terminal  in the last `n` days. Because we don't know what is the right value for `n`, we use multiple values for `n` and compute a set of features for each of them.
```

Comment:
isslation --> isolation. Double whitespace after terminal. 

Because we don't know the correct value for `n`, we use multiple values for `n` and compute a set of features for each of them.

Reply:
*Done